src/Sarif.TestUtilities/Utilities.cs

```diff
@@ -17,27 +17,6 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class Utilities
     {
-
-        // As the SARIF Technical Committee develops the SARIF specification, it
-        // releases incremental versions of the schema, with SemVer versions such as
-        // "2.0.0-csd.2.beta.2018-09-26". To avoid having to update the version strings
-        // in the test files every time this happens, we replace the version
-        // in the test file with the current version after reading the file
-        // into memory.
-        private const string VersionPropertyPattern = @"""version""\s*:\s*""[^""]+""";
-        private static readonly Regex s_VersionRegex = new Regex(VersionPropertyPattern, RegexOptions.Compiled);
-
-        private const string SchemaPropertyPattern = @"""\$schema""\s*:\s*""[^""]+""";
-        private static readonly Regex s_SchemaRegex = new Regex(SchemaPropertyPattern, RegexOptions.Compiled);
-
-        public static string UpdateVersionNumberToCurrent(string logContents)
```

Comment:
>UpdateVersionNumberToCurrent [](start = 29, length = 28)

This utility also previously replaced the run.tool version! Oops. We now use the JParser to make the sarifLog.version update.

Reply:
I hope this is the last time I ever think I can get away with using a regex to modify a JSON, XML, or HTML file.

---
In reply to: [226504993](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226504993) [](ancestors = 226504993)